### PR TITLE
Refactor clubuser.go

### LINKED ISSUE #112: Refactor clubuser db code
Refactor clubuser.go to be more like eventapplicant.go.
- informative errors 
- use proto models 
- make reusable db models
- ensure old tests still work with new changes 😈

### PR DESCRIPTION
**Pull request by srijonsaha**
:tickets: **Ticket(s)**: Closes #112

---

## :construction_worker: Changes
- clubusers.go now uses proto models
- email verification is keyed by the user email and verification hash

## :flashlight: Testing Instructions

- clubusers_test.go

### CODE REVIEW

File: protobuf/models/models.proto [hunk 1]
```diff
@@ -4,6 +4,34 @@ syntax = "proto3";
...
+
+message EmailVerification {
+  string hash = 1;
+  string email = 2;
+  google.protobuf.Timestamp expiry = 3;
+}
+
 message Event {
   string period = 1;
   string eventID = 2;
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply protobuf/models/models.proto]: another way to pass timestamps might be to use `uint32` and a standard unix timestamp:

```go
int32(time.Now().Unix())
```

File: core/database/clubuser.go [hunk 2]
```diff
@@ -8,33 +8,29 @@ import (
...
 )
 
 // AddNewUser creates a new user in the database
-func (db *Database) AddNewUser(u *model.User, e *model.EmailVerification) error {
-	verify, err := dynamodbattribute.MarshalMap(e)
+func (db *Database) AddNewUser(u *models.User, e *models.EmailVerification) error {
+	if u.Email == "" || e.Email == "" || e.Hash == "" {
+		return errors.New("Keys cannot be empty")
+	}
+	uItem, err := dynamodbattribute.MarshalMap(newDBUser(u))
```
Comment (bobheadxi) [inline core/database/clubuser.go]: 🚀

File: core/database/utils.go [hunk 3]
```diff
@@ -7,27 +7,36 @@ import (
...
+	tagPrefix          = "Tag-"
 )
 
-func prefixClubID(id string) *string {
-	return aws.String(clubPrefix + id)
-}
-
-func prefixUserEmail(email string) *string {
-	return aws.String(userPrefix + email)
+func getClubsAndUsersTable() *string {
```
Comment (bobheadxi) [inline core/database/utils.go]: it does seem a *little* weird that some of these helpers return pointers, and the others return plain strings, but not a big deal

**Comment by bobheadxi** [comment]
@leesw98 for some reason, frontend builds seem to be failing: https://travis-ci.com/ubclaunchpad/pinpoint/jobs/174968733#L679

This is happening on #125 as well, any idea why? I can't reproduce

**Comment by bobheadxi** [comment]
@srijonsaha you'll want to update this branch to include #126

File: core/verifier/verifier.go [hunk 4]
```diff
@@ -13,14 +13,14 @@ import (
...
-	Expiry time.Time
+	Expiry int64
 
 	m *mailer.Mailer
 }
 
 // New sets up verification on passed email address
 func New(email string, m *mailer.Mailer) Verifier {
-	return Verifier{Email: email, Hash: generateHash(email), Expiry: time.Now().Add(24 * 7 * time.Hour)}
+	return Verifier{Email: email, Hash: generateHash(email), Expiry: int64(time.Now().Add(24 * 7 * time.Hour).Unix())}
```
Comment (leesw98) [inline core/verifier/verifier.go]: same thing here

File: core/database/clubuser_test.go [hunk 5]
```diff
@@ -24,32 +24,35 @@ func TestDatabase_AddNewUser_GetUser(t *testing.T) {
...
 				Name:  "Bob Ross",
-				Salt:  "qwer1234",
+				Hash:  "qwer1234",
 			},
-			&model.EmailVerification{
+			&models.EmailVerification{
+				Email:  "abc@def.com",
 				Hash:   "asdf",
-				Expiry: time.Now().Add(time.Hour),
+				Expiry: int64(time.Now().Add(time.Hour).Unix()),
```
Comment (leesw98) [inline core/database/clubuser_test.go]: same thing multiple times in this file

**Pull request merged**